fe/fe-core/src/main/java/com/starrocks/sql/optimizer/dump/QueryDumpInfo.java

```diff
@@ -258,7 +260,13 @@ public void addTableCreateStmt(String tableName, String createTableStmt) {
         createTableStmtMap.put(tableName, createTableStmt);
     }
 
+    public List<String> getViewList() {
+        return viewList;
+    }
+
     public void addViewCreateStmt(String viewName, String createViewStmt) {
+        // because view may be nested, so use viewList to keep the sequence
+        viewList.add(viewName);
         createViewStmtMap.put(viewName, createViewStmt);
```

Comment:
`createViewStmtMap ` is LinkedHashMap which should insert order, why add a new `viewList`?

Reply:
no, the iterator is not ordered.